kimchi/src/prover.rs

```diff
@@ -919,8 +922,9 @@ where
         let evals = {
             let power_of_eval_points_for_chunks = [zeta_to_srs_len, zeta_omega_to_srs_len];
             &chunked_evals
+                .array()
```

Comment:
I suggest re-implementing this without the `array()`, which would make this part of the code clearer :o

Reply:
Yeah, or that is another option. I wouldn't like it to look "repetitive", since the code is the same both for zeta and zetaomega. I see the point of using the iterators there, I will think of more elegant ways for this part then.

Reply:
I think there's different ways to handle that, either create a closure and call it on both, or do something like

```rust
let CombinedEvals { z: chunked_z, zw: chunked_zw } = chunked_evals;
for chunked_evals in [z, zw] { ... }
```

I think even better is to implement most of the logic on the `CombinedEvals` type as methods.